src/common/helpers/snap-builds.js

```diff
@@ -25,43 +25,116 @@
 // };
 
 export const BuildStatusColours = {
+  BLUE: 'blue',
   GREEN: 'green',
   YELLOW: 'yellow',
   RED: 'red',
   GREY: 'grey'
 };
 
+const BuildAndPubState = {
+  'NEVER_BUILT':     createState('Never built', BuildStatusColours.GREY, false, 8),
+  'BUILDING_SOON':   createState('Building soon', BuildStatusColours.GREY, 'ellipsis', 7),
+  'IN_PROGRESS':     createState('In progress', BuildStatusColours.BLUE, 'spinner', 2),
+  'FAILED_TO_BUILD': createState('Failed to build', BuildStatusColours.RED, 'cross', 1),
+  'WONT_PUBLISH':    createState('Built, won\'t be published', BuildStatusColours.GREEN, 'tick_outlined', 6),
+  'PUBLISH_SOON':    createState('Built, publishing soon', BuildStatusColours.GREY, 'tick', 2),
+  'PUBLISH_NOW':     createState('Built, publishing now', BuildStatusColours.BLUE, 'tick', 3),
+  'PUBLISH_FAILED':  createState('Built, failed to publish', BuildStatusColours.RED, 'tick', 4),
+  'PUBLISHED':       createState('Built and published', BuildStatusColours.GREEN, 'tick_filled', 5)
+};
+
+function createState(message, colour, icon, priority) {
+  return {
+    message,
+    colour,
+    icon,
```

Comment:
Yeah, I thought I'd pencil it in as it seems like an OK place for it to go, it's a reducer so it's figuring out the state for your UI. The icon names would be better as constants or codes, but no point in finessing it now since we have no icons.